packages/go/analysis/azure/post.go

```diff
@@ -930,3 +930,159 @@ func UserRoleAssignments(ctx context.Context, db graph.Database) (*analysis.Atom
+	// Step 0: Identify each AZTenant labeled node in the database.
+	processingStats := &analysis.AtomicPostProcessingStats{}
+	tenantNodes, err := FetchTenants(ctx, db)
+	if err != nil {
+		return processingStats, err
+	}
+
+	for _, tenantNode := range tenantNodes {
+		// Read the tenantId property from the AZTenant node
+		tenantID, err := tenantNode.Properties.Get(azure.TenantID.String()).String()
+		if err != nil || len(tenantID) == 0 {
+			return processingStats, fmt.Errorf("read tenant ID property value is nil")
+		}
+
+		// Step 1 & 2: Within this tenant, find all AZRole nodes
+		//             whose tenantId matches AND where
+		//             end-user approval is required AND
+		//             at least one of the primaryApprovers props is non-null.
+		var fetchedAZRoles graph.NodeSet
+		err = db.ReadTransaction(ctx, func(tx graph.Transaction) error {
+			fetchedNodes, err := ops.FetchNodeSet(tx.Nodes().Filterf(func() graph.Criteria {
+				return query.And(
+					// Step 1: Kind = AZRole and tenantId matches
+					query.Kind(query.Node(), azure.Role),
+					query.Equals(query.NodeProperty(azure.TenantID.String()), tenantID),
+					// Step 2: isApprovalRequired == true
+					query.Equals(
+						query.NodeProperty(azure.EndUserAssignmentRequiresApproval.String()),
+						true,
+					),
+					// Step 2: primaryApprovers (user or group) is not null
+					query.Or(
+						query.IsNotNull(
+							query.NodeProperty(azure.EndUserAssignmentUserApprovers.String()),
+						),
+						query.IsNotNull(
+							query.NodeProperty(azure.EndUserAssignmentGroupApprovers.String()),
+						),
+					),
+				)
+			}))
+			if err != nil {
+				return err
+			}
+			fetchedAZRoles = fetchedNodes
+			return nil
+		})
+		if err != nil {
+			return processingStats, err
+		}
+
+		// Step 3: For each AZRole that requires approval...
+		for _, fetchedAZRole := range fetchedAZRoles {
+			// Step 3a: Read the primaryApprovers list (group GUIDs)
+			userApproversID, err := fetchedAZRole.Properties.Get(
+				azure.EndUserAssignmentUserApprovers.String(),
+			).StringSlice()
+			if err != nil {
+				return processingStats, err
+			}
+			groupApproversID, err := fetchedAZRole.Properties.Get(
+				azure.EndUserAssignmentGroupApprovers.String(),
+			).StringSlice()
+			if err != nil {
+				return processingStats, err
+			}
+			principalIDs := append(userApproversID, groupApproversID...)
+			if len(principalIDs) == 0 {
+				// Step 3b: primaryApprovers is null/empty
+				// Step 3b.i: Use tenantId from this AZRole (already have tenantID)
+				// Step 3b.ii: Find GlobalAdmin AND PrivilegedRoleAdmin roles in this tenant
+				//             and create an AZRoleApprover edge from each to this role.
+				err = db.ReadTransaction(ctx, func(tx graph.Transaction) error {
+					fetchedNodes, err := ops.FetchNodeSet(tx.Nodes().Filterf(func() graph.Criteria {
+						return query.And(
+							query.Equals(query.NodeProperty(azure.TenantID.String()), tenantID),
```

Comment:
I don't think this will return the expected results.
This is stating that the node must be multiple types, which shouldn't return any results since a node can't be all 3 of these at once.

I'd double check with research as well, but I believe this should be checking for a Role that is either GlobalAdmin or PrivilegedRoleAdmin, not where the node's kind is equal to both.